ydb/library/backup/backup.cpp

```diff
@@ -529,6 +536,115 @@ void BackupTable(TDriver driver, const TString& dbPrefix, const TString& backupP
     }
 }
 
+namespace {
+
+NView::TViewDescription DescribeView(NView::TViewClient& client, const TString& path) {
+    auto status = NConsoleClient::RetryFunction([&]() {
+        return client.DescribeView(path).ExtractValueSync();
+    });
+    VerifyStatus(status);
+    return status.GetViewDescription();
+}
+
+struct TViewQuerySplit {
+    TString ContextRecreation;
+    TString Select;
+};
+
+TViewQuerySplit SplitViewQuery(TStringInput query) {
+    // to do: make the implementation more versatile
+    TViewQuerySplit split;
+
+    TString line;
+    while (query.ReadLine(line)) {
+        (line.StartsWith("--") || line.StartsWith("PRAGMA ")
+            ? split.ContextRecreation
+            : split.Select
+        ) += line;
+    }
+
+    return split;
+}
+
+void ValidateViewQuery(const TString& query, const TString& dbPath, NYql::TIssues& accumulatedIssues) {
+    NYql::TIssues subIssues;
+    if (!NDump::ValidateViewQuery(query, subIssues)) {
+        NYql::TIssue restorabilityIssue(
+            TStringBuilder() << "Restorability of the view: " << dbPath.Quote()
+            << " storing the following query:\n"
+            << query
+            << "\ncannot be guaranteed. For more information, please consult the 'ydb tools dump' documentation."
+        );
+        restorabilityIssue.Severity = NYql::TSeverityIds::S_WARNING;
+        for (const auto& subIssue : subIssues) {
+            restorabilityIssue.AddSubIssue(MakeIntrusive<NYql::TIssue>(subIssue));
+        }
```

Comment:
Почему именно subIssues внутри одного restorabilityIssue?
- в первую очередь для того, чтобы корректно отобразить случай нескольких сломанных table ref'ов внутри одного view
- во-вторых, для более красивой печати (на мой вкус) с tab'ами перед subIssues. Смотрится в итоге вот так (здесь печать идёт из специально сломанной версии YDB CLI):

<details><summary>ydb tools dump output in case the view query is invalid</summary>
  
```text

<main>: Warning: Restorability of the view: "/Root/dedicated/view" storing the following query:
$path = "/Root/dedic" || "ated/t"; select * from $path join $path on key;
cannot be guaranteed. For more information, please consult the 'ydb tools dump' documentation.
    <main>: Warning: Please check that the table reference: "$ path" contains no evaluated expressions.
    <main>: Warning: Please check that the table reference: "$ path" contains no evaluated expressions.

```

</details>